### PR TITLE
Feature/deseng 444: Engagement Metadata API

### PR DESCRIPTION
**Pull request by NatSquared**
Issue #: N/A

*Description of changes:*

- **Feature**: Add metadata management to the API [🎟️DESENG-442](https://apps.itsm.gov.bc.ca/jira/browse/DESENG-442)
  - Add new models, services, and endpoints for metadata and metadata taxonomy
  - Add test coverage for new endpoints
  - Begin documenting new endpoints using Flask-RESTX
  - Add new endpoints to the API blueprint
  - Enabled pylint for flask and flask_sqlalchemy in vscode
  - Created a new "transactional" decorator to wrap model methods in a transaction that safely rolls back on error
  - Stub out frontend components for metadata management in preparation for the next step
- **Task**: Continue switching staff users to db-based (rather than keycloak based) authentication

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the met-public license (Apache 2.0).

### CODE REVIEW

File: met-web/src/App.tsx [hunk 1]
```diff
@@ -205,7 +205,7 @@ const App = () => {
                     backgroundColor: 'white',
                     zIndex: ZIndex.footer,
                     position: 'relative',
-                    paddingTop: '5em',
+                    // paddingTop: '5em',
```
Comment (Baelx) [inline met-web/src/App.tsx:208]: Is there a reason this was left in as a comment?

File: met-web/src/components/engagement/form/ActionContext.tsx [hunk 2]
```diff
@@ -30,9 +30,9 @@ export const ActionContext = createContext<EngagementContext>({
     handleUpdateEngagementRequest: (_engagement: EngagementFormUpdate): Promise<Engagement> => {
         return Promise.reject();
     },
-    handleCreateEngagementMetadataRequest: (_engagement: EngagementMetadata): Promise<EngagementMetadata> => {
-        return Promise.reject();
-    },
+    // handleCreateEngagementMetadataRequest: (_engagement: EngagementMetadata): Promise<EngagementMetadata> => {
```
Comment (Baelx) [inline met-web/src/components/engagement/form/ActionContext.tsx:33]: Usually with commented-out code, we have a TODO comment explaining why this is here or we remove it.

**Review by Baelx — COMMENTED** [review]
Great work on this big new feature, @NatSquared ! 

Could you please just review the commented-out code and either add TODO comments to explain why we need to keep it or remove the code.

File: met-web/src/components/engagement/form/ActionContext.tsx (hunk 2 above)
Comment (NatSquared) [reply met-web/src/components/engagement/form/ActionContext.tsx:33]: I'll try to stub these methods out more completely.

File: met-web/src/App.tsx (hunk 1 above)
Comment (NatSquared) [reply met-web/src/App.tsx:208]: Not really; aside from wondering if it was still used for something

File: met-api/src/met_api/models/db.py [hunk 3]
```diff
@@ -16,16 +19,51 @@
...
-    try:
-        yield session
-        session.commit()
-    except Exception as e:  # noqa: B901, E722
-        print(str(e))
-        session.rollback()
-        raise
+ReturnType = TypeVar('ReturnType')
+
+def transactional(db=db, autocommit=True, end_session=False
```
[the remark this answers is not in the document]
Comment (NatSquared) [reply met-api/src/met_api/models/db.py:31]: Trying to provide infrastructure for transactional methods in the Model layer going forwards and reduce boilerplate, though this solution may be overengineered for our current needs

**Review by Baelx — APPROVED** [review]
Thanks for the changes and discussion! Approved

**Pull request merged**